webapp/src/js/actions/contacts.js

```diff
@@ -66,7 +66,24 @@ angular.module('inboxServices').factory('ContactsActions',
         Auth('can_view_tasks')
           .then(() => (
             TasksForContact(selected)
-              .then(taskDocs => updateSelectedContact({ tasks: taskDocs.map(doc => doc.emission) }))
+              .then(taskDocs => {
+                const taskCounts = {};
+                taskDocs.forEach(task => {
+                  const childId = task.emission.forId;
+                  if (taskCounts[childId]) {
+                    taskCounts[childId] = taskCounts[childId] + 1;
+                  } else {
+                    taskCounts[childId] = 1;
+                  }
+                });
+                selected.children.forEach(group => {
+                  group.contacts.forEach(child => {
+                    child.taskCount = taskCounts[child.id];
+                  });
+                });
+                updateSelectedContactChildren(selected.children);
```

Comment:
Updating `selected.children` here like this might cause a race condition if something else will update the children at the same time (as far as I can tell nothing would at this time, but just to be safe). 
We're reading `selected` from state, passing it to `registerTasksListener`, which in turn calls the async `Auth` and `TasksForContact`, then overwriting the `selected.children`.
We should either grab `selected` again right before updating or have a specific action and reducer function that _only_ updates every child's `taskCount` property.

Reply:
You're absolutely right - I've pushed the code down into the contacts reducer.